client/src/HOC/WithUIContext.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { UIContextProvider } from '../UIContext';
+
+const withUIContext = (WrappedComponent) => {
```

Comment:
I'm not sure an HOC is necessary.  Can't the `<UIContextProvider>` just be returned as part of the ER component?